src/attr.rs

```diff
@@ -24,12 +26,55 @@ pub fn valid<I: Iterator<Item = char>>(chars: I) -> (usize, bool) {
     (p.pos, has_attr)
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Attr<'s> {
+    raw: CowStr<'s>,
+}
+
+impl<'s> Attr<'s> {
+    pub fn write<F: FnMut(&str) -> fmt::Result>(&self, mut write_fn: F) -> fmt::Result {
```

Comment:
Since this is part of the public API, there should probably be a doc comment explaining what it does exactly and what the purpose of the function is, i.e. split on escaped chars in order to avoid string allocation of discontinous strings.

Reply:
`AttributeValue`, `AttributeValueParts` and their public API is now documented.